### PR TITLE
Correctly set Scatter.max_size

### LINKED ISSUE #12078: [BUG] Non-circle hover glyphs permanently rendered with vline/hline mode
### Software versions

Python version      :  3.8.13 | packaged by conda-forge | (default, Mar 25 2022, 06:06:49)
IPython version     :  8.2.0
Tornado version     :  6.1
Bokeh version       :  2.4.2
BokehJS static path :  ~/miniconda3/envs/test/lib/python3.8/site-packages/bokeh/server/static
node.js version     :  (not installed)
npm version         :  (not installed)
Operating system    :  macOS-10.16-x86_64-i386-64bit

### Browser name and version

_No response_

### Jupyter notebook / Jupyter Lab version

_No response_

### Expected behavior

Hover glyphs to individually render according to hline/vline cursor hit testing, and reset when the cursor moves away.

### Observed behavior

Hover glyphs for all glyphs are rendered as soon as the cursor enters the plot and never go away.

### Example code

```Python
from bokeh.models import HoverTool
from bokeh.plotting import figure, show

p = figure()
_ = p.scatter(
    x=range(5),
    y=range(5),
    marker="circle",
    size=20,
    fill_color="steelblue",
    hover_fill_color="red",
    hover_alpha=0.1,
)
p.add_tools(HoverTool(mode='vline'))
show(p)
```

### Stack traceback or browser console output

_No response_

### Screenshots

image

### PR DESCRIPTION
**Pull request by tzhang-dev**
Fixes https://github.com/bokeh/bokeh/issues/12078 . Followed the previous PR of https://github.com/bokeh/bokeh/pull/11453 and applied to scatter view.

image

### CODE REVIEW

File: bokehjs/src/lib/models/glyphs/scatter.ts [hunk 1]
```diff
@@ -22,6 +23,17 @@ export class ScatterView extends MarkerView {
     return MultiMarkerGL
   }
 
+  override _set_data(indices: number[] | null): void {
```
Comment (mattpap) [inline bokehjs/src/lib/models/glyphs/scatter.ts]: This should be moved to `MarkerView` class in `models/glyphs/marker.ts`, as it's where `size` property is defined and where it should be handleded.

**Review by mattpap — CHANGES_REQUESTED** [review]
This PR needs a rebase to fix unrelated CI failure. This also needs some unit tests in bokehjs, specifically for `_hit_*` methods that `Scatter` glyph implements.

**Comment by tzhang-dev** [comment]
Hi @mattpap , thanks for the review and feedback. For the `_hit_*` methods, do you mean those implemented in Marker glyph? Since I found no such methods in Scatter.

Also, I did the rebase and those CI errors persist. Are they still related?

**Comment by bryevdv** [comment]
> Also, I did the rebase and those CI errors persist. Are they still related? 

I think the latest two were just two flaky jobs, I have restarted them.

**Comment by tzhang-dev** [comment]
Hi maintainer, can I get some instruction on the design intent of the `_hit_*` method and how to unit test them? I am trying to learn from the code but maybe an initial guide will help.

**Comment by bryevdv** [comment]
@yah-yuan I do believe @mattpap meant in `marker.ts` (@mattpap please confirm) 

For reference, here is a utnit test for `_hit_point` on `Rect`:

https://github.com/bokeh/bokeh/blob/9ceb505a04d60efc62f9b8de8188c6629268f791/bokehjs/test/unit/models/glyphs/rect.ts#L54-L79

File: bokehjs/src/lib/models/glyphs/marker.ts [hunk 2]
```diff
@@ -127,6 +128,17 @@ export abstract class MarkerView extends XYGlyphView {
     return new Selection({indices})
   }
 
+  override _set_data(indices: number[] | null): void {
+    super._set_data(indices)
+
+    const max_size = (() => {
+      const {size} = this
+      return size.is_Scalar() ? size.value : max((size as p.UniformVector<number>).array)
+    })()
```
Comment (mattpap) [inline bokehjs/src/lib/models/glyphs/marker.ts]: With PR #12677 merged, this block can now be replaced with `const max_size = uniforms.max(this.size)` (where `uniforms` comes from `import * as uniforms from "core/uniforms"`).

**Comment by mattpap** [comment]
> I do believe @mattpap meant in marker.ts (@mattpap please confirm)

Tests need to test a concrete implementation (`Marker` is abstract), so testing `Scatter` glyph is fine. One can also implement a dummy marker-like class in tests to avoid involving `Scatter` glyph in testing.

**Review by mattpap — APPROVED** [review]

**Comment by mattpap** [comment]
@yah-yuan, thanks.

**Pull request merged**